### PR TITLE
QE: Fix wrong return value in scp method

### PR DESCRIPTION
**Pull request by srbarrios**
## What does this PR change?

Fixing remaining issue on the new Remote Node class

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed

- [x] **DONE**

## Test coverage
ℹ️ If a major new functionality is added, it is **strongly recommended** that tests for the new functionality are added to the Cucumber test suite

- Cucumber tests were fixed

- [x] **DONE**

## Links

No ports, for this particular PR, as all will go in a unique port from qe-develop.

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

File: testsuite/features/support/remote_node.rb [hunk 1]
```diff
@@ -167,47 +167,47 @@ def wait_while_process_running(process)
...
-  def inject(local_file, remote_file)
+  def inject(test_runner_file, remote_node_file)
     if @has_mgrctl
       tmp_folder, _code = run_local('mktemp -d')
-      tmp_file = File.join(tmp_folder.strip, File.basename(local_file))
-      code, _remote = scp(local_file, tmp_file)
-      if code.zero?
-        _out, code = run_local("mgrctl cp --user #{user} #{tmp_file} server:#{remote_file}")
+      tmp_file = File.join(tmp_folder.strip, File.basename(test_runner_file))
+      success = get_target('localhost').scp(test_runner_file, tmp_file, host: self)
```
Comment (nodeg) [inline testsuite/features/support/remote_node.rb:179]: This command is basically the same as below in the `else` condition. I wonder if we do not need to use a `mgrctl` command like before? How does it know it is in the container? Is it due to the fact, we run this `.inject()` method on a server node object? Then the `localhost` value would make sense.

File: testsuite/features/support/remote_node.rb [hunk 2]
```diff
@@ -167,47 +167,47 @@ def wait_while_process_running(process)
...
+  # @param remote_node_file [String] The path in the destination.
   # @return [Integer] The exit code.
-  def inject(local_file, remote_file)
+  def inject(test_runner_file, remote_node_file)
     if @has_mgrctl
       tmp_folder, _code = run_local('mktemp -d')
-      tmp_file = File.join(tmp_folder.strip, File.basename(local_file))
-      code, _remote = scp(local_file, tmp_file)
-      if code.zero?
-        _out, code = run_local("mgrctl cp --user #{user} #{tmp_file} server:#{remote_file}")
```
Comment (nodeg) [inline testsuite/features/support/remote_node.rb:181]: Does this really need to be removed?

**Review by Bischoff — APPROVED** [review]
You're starting to discover the issues that twopence protected you against :) .

LGTM

File: testsuite/features/support/remote_node.rb (hunk 2 above)
Comment (srbarrios) [reply testsuite/features/support/remote_node.rb:181]: This line?
`        _out, code = run_local("mgrctl cp --user #{user} #{tmp_file} server:#{remote_file}")
`

File: testsuite/features/support/remote_node.rb (hunk 1 above)
Comment (srbarrios) [reply testsuite/features/support/remote_node.rb:181]: So, on the first case, we want to copy a file from the controller to the "container" **host**, and then we move this file from the "container" **host** to inside the container.
In the second case, we copy a file from the controller to the host.

Comment (srbarrios) [reply testsuite/features/support/remote_node.rb:181]: `get_target('localhost')` will return a node that will point to our controller or test runner (if you run this in your local IDE)

File: testsuite/features/support/remote_node.rb (hunk 2 above)
Comment (nodeg) [reply testsuite/features/support/remote_node.rb:181]: Yes. But I am not sure. I only saw that you kept it in the `extract()` method below.

File: testsuite/features/support/remote_node.rb (hunk 1 above)
Comment (nodeg) [reply testsuite/features/support/remote_node.rb:181]: Ok, that makes sense. I somehow thought it worked the other way around. Thanks.

**Review by vandabarata — APPROVED** [review]

File: testsuite/features/support/remote_node.rb (hunk 2 above)
Comment (srbarrios) [reply testsuite/features/support/remote_node.rb:181]: I moved it to this line:
```
      if success
        _out, code = run_local("mgrctl cp --user #{user} #{tmp_file} server:#{remote_node_file}")
```

Comment (nodeg) [reply testsuite/features/support/remote_node.rb:181]: Ah, yes. Sorry my bad. I missed that. All good then.

**Comment by srbarrios** [comment]
> =~

!! Rubocop doesn't complain about it, weird.

**Comment by nodeg** [comment]
As already discussed in Slack, yes, we can use Ruby 3.1 for now, until `net-ssh` has proper support for newer Ruby versions. One thing to keep in mind, Ruby 3.1 will already be EOL after 31.03.2025 and currently only receives security fixse.

**Pull request merged**